sockeye/data_io.py

```diff
@@ -422,30 +422,29 @@ def __init__(self,
         self.data_label = [[] for _ in self.buckets]  # type: ignore
         self.data_label_average_len = [0 for _ in self.buckets]
 
-        # Per-bucket batch sizes [num seq, num word]
+        # Per-bucket batch sizes (num seq, num word)
         # If not None, populated as part of assigning to buckets
         self.bucket_batch_sizes = bucket_batch_sizes
-        # Batch size that pairs with default bucket shape to accommodate any bucket's batch size
-        # Set with call to _populate_bucket_batch_sizes() via _assign_to_buckets()
-        self.default_bucket_batch_size = 0
 
         # assign sentence pairs to buckets
         self._assign_to_buckets(source_sentences, target_sentences)
 
         # convert to single numpy array for each bucket
         self._convert_to_array()
 
-        # "Staging area" that needs to fit any size batch we're using by total number of elements
+        # "Staging area" that needs to fit any size batch we're using by total number of elements.
+        # When computing per-bucket batch sizes, we guarantee that the default bucket will have the
+        # largest total batch size.
         self.provide_data = [
             mx.io.DataDesc(name=source_data_name,
-                           shape=(self.default_bucket_batch_size, self.default_bucket_key[0]),
+                           shape=(self.bucket_batch_sizes[-1][0], self.default_bucket_key[0]),
```

Comment:
I'd still advocate for making the elements of `bucket_batch_sizes ` a `namedtuple("bucket_batch_size", ["batch_size", "word_per_batch"])`, to make `[-1][0]` a little clearer by using `[-1].batch_size` instead.

Reply:
Good idea now that we aren't modifying the values elsewhere.